hal/src/dmac/transfer.rs

```diff
@@ -89,12 +89,16 @@
 //! transaction, that will now run uninterrupted until it is stopped.
 
 use super::{
-    channel::{AnyChannel, Busy, CallbackStatus, Channel, ChannelId, Ready},
+    channel::{AnyChannel, Busy, CallbackStatus, Channel, ChannelId, InterruptFlags, Ready},
     dma_controller::{ChId, TriggerAction, TriggerSource},
     BlockTransferControl, DmacDescriptor, DESCRIPTOR_SECTION,
 };
 use crate::typelevel::{Is, Sealed};
-use core::{mem, sync::atomic};
+use core::{
+    mem,
+    ptr::null_mut,
+    sync::atomic
+};
```

Comment:
I've become a fan of the style formatting used for `core::` here, one line per used thing, but for consistency should probably have left it with the three items on the same line.